admin/admin-service/src/main/java/io/mosip/admin/packetstatusupdater/util/RestClient.java

```diff
@@ -204,9 +204,12 @@ public <T> T getApi(String url,
 		try {
 			ResponseEntity responseEntity= (ResponseEntity) restTemplate
 					.exchange(url, HttpMethod.GET, setRequestHeader(null, null), responseType);
-			if(url.contains("datashare") && responseEntity.getHeaders().getContentType().equals(MediaType.APPLICATION_JSON)){
-				throw new MasterDataServiceException(ApplicantDetailErrorCode.DATA_SHARE_EXPIRED_EXCEPTION.getErrorCode(),
-						ApplicantDetailErrorCode.DATA_SHARE_EXPIRED_EXCEPTION.getErrorMessage());
+			if(responseEntity.getHeaders().getContentType()!=null) {
+			MediaType contextType = responseEntity.getHeaders().getContentType();
+				if(url.contains("datashare") && contextType!=null && contextType.equals(MediaType.APPLICATION_JSON)){
+					throw new MasterDataServiceException(ApplicantDetailErrorCode.DATA_SHARE_EXPIRED_EXCEPTION.getErrorCode(),
```

Comment:
Are trying to check the failure response from the datashare endpoint?

Reply:
We should create a separate method to handle datashare get endpoint and log the exact datashare error code. You can still return a general error code about data share failure.

Reply:
Updated the code as suggested